Bio/PDB/Mutate.py

```diff
@@ -0,0 +1,561 @@
+        "TYR": {"axis": ["CB", "CG"], "ref_plane": ["CA", "CB", "CG", "CD1"]},
+        "MET": {"axis": ["CB", "CG"], "ref_plane": ["CA", "CB", "CG", "SD"]},
+    },
+    "CHI3": {
+        "ARG": {"axis": ["CG", "CD"], "ref_plane": ["CB", "CG", "CD", "NE"]},
+        "GLN": {"axis": ["CG", "CD"], "ref_plane": ["CB", "CG", "CD", "OE1"]},
+        "GLU": {"axis": ["CG", "CD"], "ref_plane": ["CB", "CG", "CD", "OE1"]},
+        "LYS": {"axis": ["CG", "CD"], "ref_plane": ["CB", "CG", "CD", "CE"]},
+        "MET": {"axis": ["CG", "SD"], "ref_plane": ["CB", "CG", "SD", "CE"]},
+    },
+    "CHI4": {
+        "ARG": {"axis": ["CD", "NE"], "ref_plane": ["CG", "CD", "NE", "CZ"]},
+        "LYS": {"axis": ["CG", "CE"], "ref_plane": ["CG", "CD", "CE", "NZ"]},
+    },
+}
+
+RESIDUE_ORDER = {
+    "CYS": ["N", "CA", "C", "O", "CB", "SG"],
+    "ASP": ["N", "CA", "C", "O", "CB", "CG", "OD1", "OD2"],
+    "SER": ["N", "CA", "C", "O", "CB", "OG"],
+    "GLN": ["N", "CA", "C", "O", "CB", "CG", "CD", "NE2", "OE1"],
+    "LYS": ["N", "CA", "C", "O", "CB", "CG", "CD", "CE", "NZ"],
+    "ILE": ["N", "CA", "C", "O", "CB", "CG1", "CG2", "CD1"],
+    "PRO": ["N", "CA", "C", "O", "CB", "CG", "CD"],
+    "THR": ["N", "CA", "C", "O", "CB", "CG2", "OG1"],
+    "PHE": ["N", "CA", "C", "O", "CB", "CG", "CD1", "CD2", "CE1", "CE2", "CZ"],
+    "ASN": ["N", "CA", "C", "O", "CB", "CG", "ND2", "OD1"],
+    "GLY": ["N", "CA", "C", "O"],
+    "HIS": ["N", "CA", "C", "O", "CB", "CG", "CD2", "ND1", "CE1", "NE2"],
+    "LEU": ["N", "CA", "C", "O", "CB", "CG", "CD1", "CD2"],
+    "ARG": ["N", "CA", "C", "O", "CB", "CG", "CD", "NE", "CZ", "NH1", "NH2"],
+    "TRP": [
+        "N",
+        "CA",
+        "C",
+        "O",
+        "CB",
+        "CG",
+        "CD1",
+        "CD2",
+        "CE2",
+        "CE3",
+        "NE1",
+        "CZ2",
+        "CZ3",
+        "CH2",
+    ],
+    "ALA": ["N", "CA", "C", "O", "CB"],
+    "VAL": ["N", "CA", "C", "O", "CB", "CG1", "CG2"],
+    "GLU": ["N", "CA", "C", "O", "CB", "CG", "CD", "OE1", "OE2"],
+    "TYR": ["N", "CA", "C", "O", "CB", "CG", "CD1", "CD2", "CE1", "CE2", "CZ", "OH"],
+    "MET": ["N", "CA", "C", "O", "CB", "CG", "SD", "CE"],
+}
+
+
+def load_rotamers(rotamer_loc=DUNBRACK_FILTERED):
+    """
+    Load the Dunbrack rotamer library
+    """
+    _dunbrack = defaultdict(lambda: defaultdict(lambda: defaultdict(list)))
+    with gzip.open(rotamer_loc, "rt") as fn:
+        for line in fn:
+            if line.startswith("#"):
+                continue
+            amino_acid, angle1, angle2, prob, chi1, chi2, chi3, chi4 = line.split()
+            _dunbrack[amino_acid][int(angle1)][int(angle2)].append(
+                {
+                    "prob": float(prob),
+                    "CHI1": float(chi1),
+                    "CHI2": float(chi2),
+                    "CHI3": float(chi3),
+                    "CHI4": float(chi4),
+                }
+            )
+    return _dunbrack
+
+
+def rotation_matrix(axis, theta):
```

Comment:
You should look into `Bio.PDB.vectors`, there are functions there that probably do what you want to do here.

Reply:
Nice, I havent seen these functions, I ll update